### PR TITLE
Adding more detail to profiling recipe

### PR DESCRIPTION
**Pull request by rohan-varma**
Adding an example on how to tune ``num_worker_threads`` in RPC using the profiler. The goal is to improve this recipe as we are making RPC profiling stable in 1.8 and it would be good to have a simple example on how it can be applied to tune params that affect perf.

### CODE REVIEW

**Review by SciPioneer — APPROVED** [review]

**Comment by rohan-varma** [comment]
Preview: https://deploy-preview-1320--pytorch-tutorials-preview.netlify.app/recipes/distributed_rpc_profiling.html

File: recipes_source/distributed_rpc_profiling.rst [hunk 1]
```diff
@@ -224,78 +225,196 @@ the following:
...
-            fut.wait()
-
-        print(p.key_averages().table())
+	with profiler.profile() as p:
+	        futs = []
+	        for i in range(4):
+	            fut = rpc.rpc_async(dst_worker_name, num_workers_udf_with_ops)
+	            futs.append(fut)
+	        for f in futs:
+	            f.wait()
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:273]: wrong indent?

File: recipes_source/distributed_rpc_profiling.rst [hunk 2]
```diff
@@ -224,78 +225,196 @@ the following:
...
-      return torch.rand((3, 3), requires_grad=True)
+::
 
-    def udf_with_ops():
-      import time
-      time.sleep(1)
-      t1, t2 = random_tensor(), random_tensor()
-      torch.add(t1, t2)
-      torch.mul(t1, t2)
+	def num_workers_udf_with_ops():
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:253]: indent. Let's convert tabs into spaces.

File: recipes_source/distributed_rpc_profiling.rst [hunk 3]
```diff
@@ -224,78 +225,196 @@ the following:
...
+
+
+We see a clear ~2x speedup, and hypothesize that this speedup is due to exploiting parallelism on the server due
+to the additional cores available. However, how can we ensure that this speedup is due to the increase in cores?
+Taking a look at the trace visualization helps with this. Below is the trace when we set ``num_worker_threads=1``:
+
+.. image:: ../_static/img/oneworker.png
+   :scale: 25 %
+
+Focusing in on the trace for ``node 1``, we can see that the RPCs are ran serially on the server.
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:318]: Focusing in on -> Focusing on

File: recipes_source/distributed_rpc_profiling.rst [hunk 4]
```diff
@@ -224,78 +225,196 @@ the following:
...
+	import torch.multiprocessing as mp
+	import os
+	import logging
+	import sys
+
+	logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+	logger = logging.getLogger()
+
+	def random_tensor():
+	  return torch.rand((3, 3), requires_grad=True)
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:346]: inconsistent indents 2 vs 4

File: recipes_source/distributed_rpc_profiling.rst [hunk 5]
```diff
@@ -3,9 +3,10 @@ Profiling PyTorch RPC-Based Workloads
...
--  An overview of the `Distributed RPC Framework`_
--  An overview of the `PyTorch Profiler`_
--  How to use the profiler to profile RPC-based workloads
+-  An overview of the `Distributed RPC Framework`_.
+-  An overview of the `PyTorch Profiler`_.
+-  How to use the profiler to profile RPC-based workloads.
+-  A short example showcasing how to use the profiler to tune RPC parameters.
 
 Requirements
 ------------
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:12]: shall we update the requirements from PyTorch 1.6 to PyTorch 1.6+?

Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:12]: typo on line 123: aformentioned -> aforementioned

File: recipes_source/distributed_rpc_profiling.rst [hunk 6]
```diff
@@ -203,7 +204,7 @@ Here we can see that the user-defined function has successfully been profiled wi
 (slightly greater than 1s given the ``sleep``). Similar to the above profiling output, we can see the
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:204]: on line 152: 

due RPC -> due to RPC

**Review by mrshenli — APPROVED** [review]
Added some minor comments. LGTM!

**Comment by rohan-varma** [comment]
Thanks for the suggestions @mrshenli! Addressed all of the comments and made the spacing consistent.

**Pull request merged**